### DISCUSSION TITLE
Sending logs to external fluentd with my own tag

### DISCUSSION THREAD

**Discussion by Earsil** [kube-logging/logging-operator#1539] [Q&A] [answered]
Hi,
I have a following problem. I want to gather logs from a specific namespace and send it to external fluentd server with a tag specified by me. At first I thought that this is possible by setting tags option in output setting. But it seems that this only for grouping logs in chunk by tag. By searching more, I found that each log message is being tagged at fluentbit level. There is sth like this in config:
```
......

[INPUT]
    Name         tail
    DB  /tail-db/tail-containers-state.db
    DB.locking  true
    Mem_Buf_Limit  5MB
    Parser  docker
    Path  /var/log/containers/*.log
    Refresh_Interval  5
    Skip_Long_Lines  On
    Tag  kubernetes.*
....
```

So I assume that each log is tagged like kubernetes.* what does * add ? I couldn't find anything about it

So my main question, is it even possible to retag messages send to external fluentd ? Or maybe there is a way to somehow edit Tag in fluentbit so I can add namespace name here ?

And finally, maybe I got this all wrong and there is a simple and easy solution.

**Comment by pepov** [comment, marked as answer]
You can use the `tag_normaliser` filter like in this example:
https://github.com/kube-logging/logging-operator/issues/1117

Documentation and available options:
https://kube-logging.dev/docs/configuration/plugins/filters/tagnormaliser/

One thing to remember is that this filter should come first in the filters list, otherwise all the filters before it will be applied twice.

**Comment by Earsil** [comment]
Thank you. I somehow missed that in docs.

**Comment by Earsil** [comment]
@pepov Is it possible to tag logs with two different tags with this ? I mean one "projecttag" plus "appspecifictag"?

**Reply by pepov** [reply]
Sorry, I missed this. As far as I know fluentd doesn't support multiple tags on the logs
